packages/astro/src/core/util.ts

```diff
@@ -99,19 +99,17 @@ export function viteID(filePath: URL): string {
 }
 
 /** An fs utility, similar to `rimraf` or `rm -rf` */
-export function removeDir(_dir: URL): void {
-	const dir = fileURLToPath(_dir);
-	fs.rmSync(dir, { recursive: true, force: true });
+export function removeDir(dir: URL): void {
+	fs.rmSync(dir.toString(), { recursive: true, force: true, maxRetries: 3 });
 }
 
-export function emptyDir(_dir: URL, skip?: Set<string>): void {
-	const dir = fileURLToPath(_dir);
-	if (!fs.existsSync(dir)) return undefined;
-	for (const file of fs.readdirSync(dir)) {
+export function emptyDir(dir: URL, skip?: Set<string>): void {
+	if (!fs.existsSync(dir.toString())) return undefined;
+	for (const file of fs.readdirSync(dir.toString())) {
 		if (skip?.has(file)) {
 			continue;
 		}
-		fs.rmSync(path.resolve(dir, file), { recursive: true, force: true });
+		fs.rmSync(new URL(file, dir).toString(), { recursive: true, force: true, maxRetries: 3 });
```

Comment:
v14 doesn't support file URLs in Node `fs` methods. This is why we used `fileURLToPath(_dir)` in these functions.

Reply:
Testing this theory here: https://github.com/withastro/astro/pull/2761

Reply:
The Node docs lead me astray! I changed this after just `maxRetries` didn't fix it :(